lib/wibox/drawable.lua

```diff
@@ -407,13 +408,8 @@ function drawable.new(d, widget_context_skeleton, drawable_name)
 
     -- Set up our callbacks for repaints
     ret._redraw_callback = function(hierar, arg)
-        -- XXX: lgi will lead us into memory-corruption-land when we use an
-        -- object after it was GC'd. Try to detect this situation by checking if
-        -- the drawable is still valid. This is only a weak indication, but it
-        -- seems to be the best that we can do. The problem is that the drawable
-        -- could not yet be GC'd, but is pending finalisation, while the
-        -- cairo.Region below was already GC'd. This would still lead to corruption.
-        if not ret.drawable.valid then
+        -- Avoid crashes when a drawable was partly finalized and dirty_area is broken.
+        if not ret._visible then
```

Comment:
Should something similar also be done in `_layout_callback` to avoid relayouts of partly finalised widgets?
Something like the following:
```diff
@@ -426,7 +434,9 @@ function drawable.new(d, widget_context_skeleton, drawable_name)
             return
         end
         ret._need_relayout = true
-        ret:draw()
+        if ret._visible then
+            ret:draw()
+        end
     end
 
     -- Add __tostring method to metatable.
```
(If the drawable becomes visible again, the relayout will be done. If it never becomes visible again, the relayout will just keep being pending until the drawable is garbage collected.)